providers/jldap/src/java/edu/amc/sakai/user/UserAttributeRoleProvider.java

```diff
@@ -60,8 +57,10 @@ public void init() {
 					}
 				}
 			} catch (UserNotDefinedException e) {
+				// TODO What about caching failed user lookups because the user isn't found.
```

Comment:
Yes, I believe I contributed a negative lookup cache at some point. The tough part is determining if LDAP was unavailable or if the individual lookup failed